linker/shared/src/test/scala/org/scalajs/linker/LinkerTest.scala

```diff
@@ -50,6 +52,39 @@ class LinkerTest {
     testLink(helloWorldClassDefs, MainTestModuleInitializers)
   }
 
+  /** Makes sure that linking a "substantial" program (using `println`) does
+   *  not trigger the second attempt in the Emitter due to dangerous global
+   *  refs.
+   */
+  @Test
+  def linkNoSecondAttemptInEmitter(): AsyncResult = await {
+    val PrintStreamClass = ClassName("java.io.PrintStream")
+    val PrintStreamType = ClassType(PrintStreamClass)
+
+    val classDefs = List(
+      mainTestClassDef {
+        val err = ApplyStatic(EAF, "java.lang.System",
+            m("err", Nil, ClassRef(PrintStreamClass)), Nil)(PrintStreamType)
+        Apply(EAF, err, m("println", List(O), V), List(str("Hello world!")))(NoType)
+      }
+    )
```

Comment:
Just use this instead? https://github.com/scala-js/scala-js/blob/92c64df83377d47c55d3de0ff78d10da3ba06bcc/linker/shared/src/test/scala/org/scalajs/linker/OptimizerTest.scala#L179-L183